source/faucet/main.d

```diff
@@ -382,6 +382,15 @@ public class Faucet : FaucetAPI
         auto owned_utxo_rng = this.state.owned_utxos.byKeyValue()
             // do not pick already used UTXOs
             .filter!(pair => pair.key !in this.used_utxos);
+
+        auto owned_utxo_len = owned_utxo_rng.walkLength;
+        assert(owned_utxo_len >= 0);
+        if (owned_utxo_len <= 1)
+        {
+            logError("Insufficient UTXOs in storage. # of UTXOs: %s", owned_utxo_len);
+            throw new Exception(format("Insufficient UTXOs in storage. # of UTXOs: %s", owned_utxo_len));
+        }
+
         auto first_utxo = owned_utxo_rng.front;
```

Comment:
Perhaps consider using `take(1)` and then check the length (take will take up to the given number without assertion or throw). If it is zero then throw.

Reply:
Good idea, but I could not check the length of `owned_utxo_rng` without calling `walkLength`. The range does not accept `length` or `size`.

Reply:
I meant to check the length of the result of take(1).